### DISCUSSION TITLE
capacitor v3, android 7.1.2, plugins doesn't work

### DISCUSSION THREAD

**Discussion by kontul-dev** [ionic-team/capacitor#4638] [Q&A] [answered]
I have simple react app, where I updated Capacitor to v3, and added some capacitor plugins.
App woks in emulator, and on Android 9 but not on Android 7.1.2

I also created separated app, from scratch, using react-create-app, Capacitor v3, only capacitor plugins - same problem.

To be exact, this is state on Android 7.1.2:
1. None of @capacitor/ plugins seem to work (all those I added)
2. When I call getPlatforms() from @ionic/react, "cordova" is not listed (it is on Android 9, in Capacitor v2 and v3, as well as on Android 7 + Capacitor v2)
3. other plugin that I use, phonegap-plugin-barcodescanner, doesn't work either (it works on Androind 9 + Capacitor 2/3, as well as on Android 7 + Capacitor v2)

Any ideas, what could cause this behavior?

package.json:

> {
  "name": "bc-123kurier",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@capacitor/android": "^3.0.0",
    "@capacitor/cli": "^3.0.0",
    "@capacitor/core": "^3.0.0",
    "@capacitor/device": "^1.0.0",
    "@capacitor/dialog": "^1.0.0",
    "@capacitor/toast": "^1.0.0",
    "@ionic-native/barcode-scanner": "^5.33.0",
    "@ionic-native/core": "^5.33.0",
    "@ionic-native/device": "^5.33.0",
    "@ionic/react": "^5.6.7",
    "cordova-plugin-device": "^2.0.3",
    "node-sass": "^6.0.0",
    "phonegap-plugin-barcodescanner": "^8.1.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-router-dom": "^5.2.0",
    "react-scripts": "4.0.3",
    "rxjs": "^7.0.1",
    "typescript": "^4.1.2",
    "web-vitals": "^1.0.1"
  },
  "devDependencies": {
    "@types/node": "^12.0.0",
    "@types/react": "^17.0.0",
    "@types/react-dom": "^17.0.0",
    "typescript-plugin-css-modules": "^3.2.0"
  },
  "scripts": {
    "start": "set PORT=5132 && react-scripts start",
    "build": "react-scripts build"
  },
  "commentedOutScripts": {
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}

**Comment by jcesarmobile** [comment, marked as answer]
can you check the userAgent of the emulator?
Capacitor 3 requires chromium/chrome 60 and some emulators use older versions and can't update the webView, so you'll have to test on real devices.

**Reply by kontul-dev** [reply]
I might put it unclear: It always works in emulator. But, I am testing it on two devices, one has Android 9, second one Android 7.1.2.
Problem is, that that second one is in fact target device (meaning app needs to work there, on that particular device).

I read UserAgent of that device, and Chrome version is 52.xxx

Does it mean, that it't not possible to use Capacitor 3 on older Android systems? Or is there something that I can do about it?

**Reply by jcesarmobile** [reply]
In android 7 the webview version depends on the chrome version, if you update chrome then it should work

**Reply by kontul-dev** [reply]
Ok, thanks for answer. Unfortunately I can't (re)install Chrome on that device. It is POS terminal device, with limited access to system.
I'll use capacitor v2 for this project.
